compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -21,6 +22,61 @@ use super::{
 };
 use crate::ty;
 
+/// Functionality required for the bytes of an `Allocation`.
+pub trait AllocBytes:
+    Clone
+    + core::fmt::Debug
+    + Eq
+    + PartialEq
+    + PartialOrd
+    + Ord
+    + core::hash::Hash
```

Comment:
Please remove the `Ord` bounds, they should not be needed. No idea why `Allocation` derives `PartialOrd`, but https://github.com/rust-lang/rust/pull/104933 removes that.